instrument/global.go

```diff
@@ -0,0 +1,61 @@
+package instrument
+
+import (
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/segmentio/analytics-go.v3"
+)
+
+var globalLock sync.Mutex
+var globalClient client = MockClient{}
+
+func setGlobalClient(client client) {
+	if client == nil {
+		return
+	}
+	globalLock.Lock()
+	globalClient = client
+	globalLock.Unlock()
+}
```

Comment:
So minor things, but by hiding this method in the package, you make it so that people can't inject their own clients to the global space (e.g. they want to instrument the existing one). 

I think exporting this (and the `client` interface) might be good. Yes I know this is kinda conflicting my earlier comment, but having a global methods like you've got here are the real benefit I was trying to allude to

Same with `getGlobalClient` ~ as you hid the actually global client you can't save it off in tests. In the featureflag we do a lot of 
```
func TestSomething (t *testing.T) {
  og := featureflag.GetGlobalClient()
  t.Cleanup(func(){ featureflag.SetGlobalClient(og) })
  featureflag.SetGlobalClient(mymockclient)
}
```

So I'd say:
- export the `client` interface (not the global value) and the methods. 
- export the `SetGlobalClient` and `GetGlobalClient`

Reply:
🖖